test/test_optim.py

```diff
@@ -320,6 +320,35 @@ def _test_complex_optimizer(self, optimizer_constructor):
 
             self.assertEqual(torch.view_as_real(complex_param), real_param)
 
+    def _test_complex_2d(self, optimizer_constructor, f=None):
+        if f is None:
+            f = rosenbrock
+        a1 = torch.randn(2, dtype=torch.complex64, requires_grad=True)
+        a1_real = a1.real.clone().detach()
+        a1_imag = a1.imag.clone().detach()
+        a1_real.requires_grad_()
+        a1_imag.requires_grad_()
+        a2 = torch.complex(a1_real, a1_imag)
+        optim1 = optimizer_constructor([a1])
+        optim2 = optimizer_constructor([a1_real, a1_imag])
+
+        for i in range(10):
+            optim1.zero_grad()
+            optim2.zero_grad()
+            a2 = torch.complex(a1_real, a1_imag)
+            f(a1).backward()
+            f(a2).backward()
+
+            assert(torch.allclose(a1.grad.real, a1_real.grad))
```

Comment:
Please use the builtin asserts on `self.` to play nicely with the test suite in general.

Reply:
:+1: